Engine/Setup/BaseSetupHandler.cs

```diff
@@ -54,12 +54,6 @@ public static class BaseSetupHandler
             var historyRequests = new List<HistoryRequest>();
             foreach (var cash in cashToUpdate)
             {
-                // if we already added a history request for this security, skip
-                if (historyRequests.Any(x => x.Symbol == cash.ConversionRateSecurity.Symbol))
```

Comment:
Good catch, this shouldn't happen, if two cash share the same conversion rate security it should be the same cash symbol, which isn't a valid scenario